lib/web/desktop.go

```diff
@@ -371,3 +380,81 @@ func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
 	}
 	return trace.NewAggregate(retErrs...)
 }
+
+// createCertificateBlob creates Certificate BLOB
+// It has following structure:
+// 	CertificateBlob {
+// 		PropertyID: u32, little endian,
+// 		Reserved: u32, little endian, must be set to 0x01 0x00 0x00 0x00
+// 		Length: u32, little endian
+// 		Value: certificate data
+// 	}
+//
+func createCertificateBlob(certData []byte) []byte {
+	buf := new(bytes.Buffer)
+	// PropertyID for certificate is 32
+	binary.Write(buf, binary.LittleEndian, int32(32))
+	binary.Write(buf, binary.LittleEndian, int32(1))
+	binary.Write(buf, binary.LittleEndian, int32(len(certData)))
+	buf.Write(certData)
+
+	return buf.Bytes()
+}
+
+func (h *Handler) desktopAccessScriptConfigureHandle(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+	httplib.SetScriptHeaders(w.Header())
+
+	clusterName, err := h.auth.proxyClient.GetDomainName(r.Context())
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	certAuthority, err := h.auth.proxyClient.GetCertAuthority(
+		r.Context(),
+		types.CertAuthID{Type: types.UserCA, DomainName: clusterName},
+		false,
+	)
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if len(certAuthority.GetActiveKeys().TLS) != 1 {
+		return nil, trace.BadParameter("expected one TLS key pair, got %v", len(certAuthority.GetActiveKeys().TLS))
+	}
+
+	keyPair := certAuthority.GetActiveKeys().TLS[0]
+	block, _ := pem.Decode(keyPair.Cert[:len(keyPair.Cert)-1])
```

Comment:
Great question! At first I thought that the Windows is truncating the last byte in the import process. I couldn't make to work the auto-import without removing that last byte when I was importing the certificate. So I manually tested it and when I imported certificate, exported it and then compared these two, they where different by that last byte. 

Your question got me thinking and I've re-checked everything and I think I found what is the real issue.

The cerficiate has been exported using this command:
```
tctl auth export --type=windows > exported.cer
```

When you take a look into hexdump of the certificates, especially on the last byte:

```
❯ hexdump exported.cer

[ ... remaining cert data ... ]
0000370 c32d 37a5 a11e 8905 241d d8c5 38f3 6e10
0000380 5c95 061d 4449 ead8 dfd6 9757 a49e 8acb
0000390 0afe
0000392
```

```
❯ hexdump exported-and-imported.cer
[ ... remaining cert data ... ]
0000370 c32d 37a5 a11e 8905 241d d8c5 38f3 6e10
0000380 5c95 061d 4449 ead8 dfd6 9757 a49e 8acb
0000390 00fe
0000391
```

You'll see that they are different: `0a` vs `00`.

Then I checked the [source code](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/auth_command.go#L293) of the auth command that was used to export the certificate and found that it is using the `fmt.Println` function to print contents of the bytes onto the stdout. It adds one extra byte to our certificate, which is new line, and the ascii code in hex for new line is `a` which is exactly the difference in the hexdump output. 

When you load the exported certificate into this cool decoder for asn.1 here: https://lapo.it/asn1js/ it will show at the top of the page that the input contains 1 more byte to decode. 

When I've changed `fmt.Println` to `fmt.Print` in the auth command then the manual process started to working without removing the last byte. 

It turns out that removing last byte is not needed when doing it here so I'll remove that part. 
The reason it worked when I was incorrectly removing the last byte from valid certificate is that for some reason PEM decoder produces valid DER representation even without that last NULL byte 😄 

Also, we might consider changing the auth command so it does not contains extra new line byte.

Reply:
Good find. I have actually seen this before!

You're right, we should not print the \n character for --type=windows since it's a binary file as opposed to PEM.